modules/hdinsight-cluster/main.tf

```
@@ -41,6 +40,21 @@ resource "azurerm_hdinsight_hbase_cluster" "hdinsight_hbase_cluster" {
       subnet_id             = var.subnet_id
       virtual_network_id    = var.vnet_id
       target_instance_count = var.worker_count
+
+      # Schedule based auto-scaling
+      autoscale {
+        recurrence {
+          timezone = var.scaling_timezone
+          dynamic "schedule" {
+            for_each = var.scaling_times
+            content {
+              days                  = var.scaling_days
+              time                  = schedule.value
+              target_instance_count = var.scaled_target_instance_counts[schedule.key]
```

scaled_target_instance_counts and scaling_times are indirectly coupled.

Maintaining the two lists obviously works, but would it make sense to combine these two thins into a map since they're coupled? It may decrease the chance of someone not keeping the two lists in sync.

Something like:
```
{"13:00": 3, "14:00", 4}
```

You should be able to reference both the key and value in the dynamic block:
```suggestion
            content {
              days                  = var.scaling_days
              time                  = schedule.key
              target_instance_count = schedule.value
```


Maybe something like wild like this where we inject the days in that time too:
```
{"13:00": {["Monday","Tuesday"]: 3, "14:00", {["Monday", "Friday"]: 4}
```

Then you could do something like this:
```suggestion
            content {
              days                  = var.scaling_times[schedule.key].key
              time                  = schedule.key
              target_instance_count = var.scaling_times[schedule.key].value
```

Obviously haven't thought this through and I don't 💯  remember the constraints, but maybe there's something there.